adjust FocusWidget: GoToItem emulates Space key in WSearchRelatedTracksMenu

ronso0 opened the pull request:
Requested here https://github.com/mixxxdj/mixxx/pull/12213#issuecomment-2128069547

This allows to toggle the search actions and click the Search button with `[Library],GoToItem`, which would previously activate the actions (trigger search and close the menu).

Needs some testing with controllers !

acolombier commented:
I'll give it a go ASAP!

acolombier reviewed (commented):
Unfortunatelly, it doesn't work with my controller (S4Mk3, tested with the mapping available on that same branch)

Here is the logic snippet used for testing:

```js
const currentlyFocusWidget = engine.getValue("[Library]", "focused_widget");
// 3 == Tracks table or root views of library features
if (this.shifted && currentlyFocusWidget === 0) {
    script.triggerControl("[Playlist]", "ToggleSelectedSidebarItem");
} else if (currentlyFocusWidget === 3 || currentlyFocusWidget === 0) {
    script.triggerControl(this.group, "LoadSelectedTrack");
} else {
    script.triggerControl("[Library]", "GoToItem");
}
```

However, instead of selecting the checkbox, it just trigger the search with that unique criteria. That seems expected reading the [GoToItem](https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[Library]-GoToItem) doc. Perhaps we should add another value to `GoToItem`? (1 == Enter, 2 == Space)

Note that `focused_widget` value was always `4` so something might be wrong with the newly introduced `FocusWidget::SearchRelatedMenu`.

ronso0 commented:
Admittedly this was just a quick shot, I didn't even test it :grimacing: 

Turns out only the **main** popup is considered, not its active child WSearchRelatedTracksMenu.
And when cycling through the search actions, the reported focused widget is first the WSearchRelatedTracksMenu, then the QCheckBox.

ronso0 commented:
@acolombier This should work now, at least `LibraryControl::getFocusedWidget()` now properly detects the search menu.

ronso0 committed:
adjust FocusWidget: [Library],GoToItem uses space in WSearchRelatedTracksMenu

daschuer commented on src/library/librarycontrol.cpp:
```diff
@@ -866,6 +811,9 @@ FocusWidget LibraryControl::getFocusedWidget() {
 
+    qWarning() << "     .";
+    qWarning() << "     focWin:" << focusWindow;
+    qWarning() << "       name:" << focusWindow->objectName();
```
This seems to be debug leftover.

daschuer commented on src/library/librarycontrol.cpp:
```diff
@@ -876,7 +824,18 @@ FocusWidget LibraryControl::getFocusedWidget() {
         // QComboBoxPrivateContainerClassWindow
         //    = QComboBoxListView of WEffectSelector, WSearchLineEdit, ...
-        return FocusWidget::ContextMenu;
+        auto* focWid = QApplication::focusWidget();
```
```suggestion
        auto* pFocWid = QApplication::focusWidget();
```

daschuer reviewed (commented):
The code looks good, except of the bug leftovers.

ronso0 commented:
Added the p prefix and removed the trace commit.
Will do a test with my controller tomorrow and report back (simulation with mapping for MIDI Through already worked)

daschuer reviewed (approved):
LGTM, Thank you.
